charts/argo-cd/templates/argocd-application-controller/deployment.yaml

```diff
@@ -0,0 +1,108 @@
+{{- if .Values.controller.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "argo-cd.controller.fullname" . }}
```

Comment:
What if we don't template this and use it like in the labels below: `{{ include "argo-cd.fullname" . }}-{{ .Values.controller.name }}` We get rid of one extra level of indirection, have less code (we should remove the functions from `_helpers.tpl`) and the name becomes more clear.
This applies to all the naming functions used.

Reply:
That is an option but I would argue that it becomes harder to make global changes to names in the future.  having the name of objects defined in the helpers.tpl is very standard and if in the future these names need to change, enabling that change to occur in a single place is a large benefit.

if changed you would need to accurately update names:
argo controller: 11 places / 9 files
argo server: 14 places / 11 files
reposerver: 9 places / 7 files

the way it is now, you simply need to update 3 lines in 1 file.